src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
```

Comment:
I agree. I misremembered what [async does](https://github.com/caolan/async#seriestasks-callback) and thought callbacks were required. Turns out optional callbacks is a normal thing.